### DISCUSSION TITLE
More informative content in warning and error messages

### DISCUSSION THREAD

**Discussion by lucpotage** [nuxt/nuxt#19600] [General]
As a beginner, I find it hard to understand Nuxt 3 errors due to the lack of insights in the error messages.
Last example I have is the following:

> [Vue warn]: Unhandled error during execution of setup function at <Index onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< undefined > >

I have no idea what this means and how to solve this issue. There is no link to the docs to better understand what’s going wrong. I just know that this happens after a force refresh with server rendering but not in the browser, but why? I just use a useFetch call in a composable, I thought it would work on server side.

I started my Journey with Nuxt 1 and Nuxt 2. I must admit that upgrading to Nuxt 3 is not simple. I cannot say it would be better with another similar framework, just that I feel frequently lost, and I can waste several hours without any clue for basic stuff. I know I can share a repo but most of the time, I simply can’t because I’m not ready to share it with the public so I’m stuck.

It would be very insightful to have at least one long video explaining the different rendering modes, what to pay attention to in the code, in composables, plugins, helpers, layouts, pages, components, etc.

I wish I could contribute myself to that but right now I feel like a newbie with Nuxt 3 whereas I was confident with Nuxt 2 possibilities, and despite the great docs.

**Comment by lucpotage** [comment]
I finally found the solution. I was wrongly using `.value` on data from `$fetch` whereas it seems I don’t have too. The warning message did not help me, only a lot of console.log in my code to understand what is passed in promises.

**Comment by TheAlexLichter** [comment]
>  I know I can share a repo but most of the time, I simply can’t because I’m not ready to share it with the public so I’m stuck.

This is often a wrong belief. You can cut down your code to only include the error you receive and as less logic as possible. Also, APIs can be replaced with placeholder APIs like https://api.nuxtjs.dev/mountains

> I have no idea what this means and how to solve this issue. There is no link to the docs to better understand what’s going wrong.

In this case, that's not very trivial as lots of errors could be triggered in a `<script setup>` top level.
But - error messages can hopefully be improved in the future 🙏🏻

> It would be very insightful to have at least one long video explaining the different rendering modes, what to pay attention to in the code, in composables, plugins, helpers, layouts, pages, components, etc.

I agree, more info in the docs would be very helpful!
